mpcs/src/basefold/query_phase.rs

```diff
@@ -157,239 +134,172 @@ pub fn batch_verifier_query_phase<E: ExtensionField, Spec: BasefoldSpec<E>>(
+                // fold and query
+                let mut cur_num_var = max_num_var;
+                let mut log2_height = cur_num_var + log2_blowup - 1;
+                // -1 because for there are only #max_num_var-1 openings proof
+                let rounds = cur_num_var - S::get_basecode_msg_size_log() - 1;
 
-            // first folding challenge
-            let r = fold_challenges.first().unwrap();
+                assert_eq!(rounds, fold_challenges.len() - 1);
+                assert_eq!(rounds, proof.commits.len(),);
+                assert_eq!(rounds, opening_ext.len(),);
 
-            let mut folding_sorted_order_iter = folding_sorted_order.iter();
-            // take first batch which num_vars match max_num_var to initial fold value
-            let mut folded = folding_sorted_order_iter
-                .by_ref()
-                .peeking_take_while(|(num_vars, _)| **num_vars == cur_num_var)
-                .map(|(_, index)| {
-                    let (lo, hi) = &base_codeword_lo_hi[*index];
-                    let coeff =
-                        <Spec::EncodingScheme as EncodingScheme<E>>::verifier_folding_coeffs(
-                            vp,
-                            cur_num_var
-                                + <Spec::EncodingScheme as EncodingScheme<E>>::get_rate_log()
-                                - 1,
-                            idx,
-                        );
-                    codeword_fold_with_challenge(&[*lo, *hi], *r, coeff, inv_2)
-                })
-                .sum::<E>();
+                // first folding challenge
+                let r = fold_challenges.first().unwrap();
+                let coeff = S::verifier_folding_coeffs(vp, log2_height, idx);
+                let (lo, hi) = reduced_openings[&log2_height];
+                let mut folded = codeword_fold_with_challenge(&[lo, hi], *r, coeff, inv_2);
 
-            let mut n_d_i = n_d_next;
-            for (
-                (pi_comm, r),
-                CommitPhaseProofStep {
-                    sibling_value: leaf,
-                    opening_proof: proof,
-                },
-            ) in commits
-                .iter()
-                .zip_eq(fold_challenges.iter().skip(1))
-                .zip_eq(opening_ext)
-            {
-                cur_num_var -= 1;
+                for (
+                    (pi_comm, r),
+                    CommitPhaseProofStep {
+                        sibling_value,
+                        opening_proof: proof,
+                    },
+                ) in proof
+                    .commits
+                    .iter()
+                    .zip_eq(fold_challenges.iter().skip(1))
+                    .zip_eq(opening_ext)
+                {
+                    cur_num_var -= 1;
+                    log2_height -= 1;
 
-                let is_interpolate_to_right_index = (idx & 1) == 1;
-                let new_involved_codewords = folding_sorted_order_iter
```

Comment:
The sorting clean up is awesome and make overall flow much much simpler!
My side question: after sorted was remove, could you elaborate how it was replaced, such as a query with num_var < current_num_var could be involved into later round folding correctly?